configuration/example.config.toml

```diff
@@ -0,0 +1,200 @@
+near_archival_rpc_url = "https://beta.rpc.mainnet.near.org" # Optional
+
+# Rpc server general configuration
+[general.rpc_server]
+
+# Referer header value
+# We want to set a custom referer to let NEAR JSON RPC nodes know that we are a read-rpc instance
+referer_header_value = "read-rpc"
+
+# Port for RPC server
+port = 8000
+
+# Max gas burnt for contract function call
+# Default value is 300_000_000_000_000
+max_gas_burnt = 300_000_000_000_000
+
+# Max available memory for `block_cache` and `contract_code_cache` in gigabytes
+# By default we use all available memory
+limit_memory_cache = 0
+
+# Reserved memory for running the application in gigabytes
+# By default we use 0.25 gigabyte (256MB or 268_435_456 bytes)
+reserved_memory = 0.25
+
+# Block cache size in gigabytes
+# By default we use 0.125 gigabyte (128MB or 134_217_728 bytes)
+# One cache_block size is ≈ 96 bytes
+# In 128MB we can put 1_398_101 cache_blocks
+block_cache_size = 0.125
+
+# Tx indexer general configuration
+[general.tx_indexer]
+
+# Indexer ID to handle meta data about the instance
+indexer_id = "tx-indexer" # Unique indexer ID
+
+# Port for metrics server
+port = 8080
+
+# To restore cache from db we use smart range blocks
+# Regular transaction takes some blocks to be finalized
+# We don't need to restore too old transactions for the indexer because we will probably never be able to reassemble them.
+# We use a range of 1000 blocks for our peace of mind. We also leave the option to increase or decrease this range
+cache_restore_blocks_range = 1000
+
+# State indexer general configuration
+[general.state_indexer]
+
+# Indexer ID to handle meta data about the instance
+indexer_id = "state-indexer" # Unique indexer ID
+
+# Port for metrics server
+port = 8081
+
+# Concurrency for state indexer
+concurrency = 1
+
+# Epoch indexer general configuration
+[general.epoch_indexer]
+
+# Indexer ID to handle meta data about the instance
+indexer_id = "epoch-indexer" # Unique indexer ID
+
+# Tracking acconunts configuration
+[accounts]
+# Accounts to track. By default we track all accounts.
+# You can specify a list of accounts to track.
+#tracked_accounts = ["test.near"]
+tracked_accounts = []
+
+# Tracking state changes configuration
+[state_changes]
+# State changes to track. By default we track all state changes.
+# You can specify a list of state changes to track.
+# Possible values: "state", "access_key", "contract_code"
+# "accounts" are tracked from the [accounts] section
+tracked_changes = ["State", "AccessKey", "ContractCode"]
+
+# Lake framework configuration
+[lake_config]
+
+# Lake framework AWS access key id
+aws_access_key_id = "${AWS_ACCESS_KEY_ID}"
+
+# Lake framework AWS secret access key
+aws_secret_access_key = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+
+# Lake framework AWS default region
+aws_default_region = "eu-central-1"
+
+# Lake framework bucket name
+aws_bucket_name = "near-lake-data-mainnet"
+
+# Database configuration
+[database]
```

Comment:
Could you add common parameters (like `preferred_db,` `max_retry,` `keepalive_interval,` etc.) here with a way of overriding the common ones for specific services, please?

Reply:
I don't think we need to specify a common configuration since we already have default values.

Reply:
It's not about the defaults, but rather to be able to set a custom parameters to everyone and override for specific service.